modules/apps/callbacks/ica_test.go

```diff
@@ -0,0 +1,247 @@
+			types.CallbackTriggerAcknowledgementPacket,
+			true,
+		},
+		{
+			"success: source callback with other json fields",
+			fmt.Sprintf(`{"src_callback": {"address": "%s"}, "something_else": {}}`, callbackAddr),
+			types.CallbackTriggerAcknowledgementPacket,
+			true,
+		},
+		{
+			"success: source callback with malformed json",
+			fmt.Sprintf(`{"src_callback": {"address": "%s"}, malformed}`, callbackAddr),
+			"none",
+			true,
+		},
+		{
+			"success: source callback with missing address",
+			`{"src_callback": {"address": ""}}`,
+			"none",
+			true,
+		},
+		{
+			"failure: dest callback with low gas (error)",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s", "gas_limit": "450000"}}`, callbackAddr),
+			"none",
+			false,
+		},
+		{
+			"failure: source callback with low gas (error)",
+			fmt.Sprintf(`{"src_callback": {"address": "%s", "gas_limit": "450000"}}`, callbackAddr),
+			types.CallbackTriggerAcknowledgementPacket,
+			false,
+		},
+		{
+			"failure: dest callback with low gas (panic)",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s", "gas_limit": "350000"}}`, callbackAddr),
+			"none",
+			false,
+		},
+		{
+			"failure: source callback with low gas (panic)",
+			fmt.Sprintf(`{"src_callback": {"address": "%s", "gas_limit": "350000"}}`, callbackAddr),
+			types.CallbackTriggerAcknowledgementPacket,
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		s.Run(tc.name, func() {
+			icaAddr := s.SetupICATest()
+
+			s.ExecuteICATx(icaAddr, tc.icaMemo, 1)
+			s.AssertHasExecutedExpectedCallback(tc.expCallback, tc.expSuccess)
+		})
+	}
+}
+
+func (s *CallbacksTestSuite) TestICATimeoutCallbacks() {
+	// ICA channels are closed after a timeout packet is executed
+	testCases := []struct {
+		name        string
+		icaMemo     string
+		expCallback types.CallbackTrigger
+		expSuccess  bool
+	}{
+		{
+			"success: transfer with no memo",
+			"",
+			"none",
+			true,
+		},
+		{
+			"success: dest callback",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s"}}`, callbackAddr),
+			"none",
+			true,
+		},
+		{
+			"success: source callback",
+			fmt.Sprintf(`{"src_callback": {"address": "%s"}}`, callbackAddr),
+			types.CallbackTriggerTimeoutPacket,
+			true,
+		},
+		{
+			"success: dest callback with low gas (error)",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s", "gas_limit": "450000"}}`, callbackAddr),
+			"none",
+			true,
+		},
+		{
+			"failure: source callback with low gas (error)",
+			fmt.Sprintf(`{"src_callback": {"address": "%s", "gas_limit": "450000"}}`, callbackAddr),
+			types.CallbackTriggerTimeoutPacket,
+			false,
+		},
+		{
+			"success: dest callback with low gas (panic)",
+			fmt.Sprintf(`{"dest_callback": {"address": "%s", "gas_limit": "350000"}}`, callbackAddr),
+			"none",
+			true,
+		},
+		{
+			"failure: source callback with low gas (panic)",
+			fmt.Sprintf(`{"src_callback": {"address": "%s", "gas_limit": "350000"}}`, callbackAddr),
+			types.CallbackTriggerTimeoutPacket,
+			false,
+		},
```

Comment:
would be good to add some in-line comments to (error) and (panic) source callback testcases with a short description of what incurs a panic vs error here - right now its maybe a little arbitrary with the gas limit numbers to differentiate whats going on. 

similar to other comments if `dest callback` testcase could be removed I'd be in favour. we can just add a godoc string to the testcase explaining icahost callbacks are not supported.

Reply:
Ok. I'll only leave `"success: dest callback"` test case and remove other dest callback test cases for ica